rest/rest-server/src/main/java/org/apache/brooklyn/rest/resources/ApplicationResource.java

```diff
@@ -266,27 +268,48 @@ protected Response createFromAppSpec(ApplicationSpec applicationSpec) {
     @Override
     public Response createFromYaml(String yaml) {
         // First of all, see if it's a URL
-        URI uri;
+        Preconditions.checkNotNull(yaml, "Blueprint cannot be null");
+        URI uri = null;
         try {
-            uri = new URI(yaml);
+            String yamlUrl = yaml.trim();
+            if (Urls.isUrlWithProtocol(yamlUrl)) {
+                uri = new URI(yamlUrl);
+            }
         } catch (URISyntaxException e) {
             // It's not a URI then...
             uri = null;
         }
         if (uri != null) {
             log.debug("Create app called with URI; retrieving contents: {}", uri);
-            yaml = ResourceUtils.create(mgmt()).getResourceAsString(uri.toString());
+            try {
+                yaml = ResourceUtils.create(mgmt()).getResourceAsString(uri.toString());
+            } catch (Exception e) {
+                Exceptions.propagateIfFatal(e);
+                throw new UserFacingException("Cannot resolve URL: "+uri, e);
+            }
         }
 
         log.debug("Creating app from yaml:\n{}", yaml);
-        EntitySpec<? extends Application> spec = createEntitySpecForApplication(yaml);
+        EntitySpec<? extends Application> spec;
+        try {
+            spec = createEntitySpecForApplication(yaml);
+        } catch (Exception e) {
+            Exceptions.propagateIfFatal(e);
+            Throwable root = Throwables.getRootCause(e);
+            if (root instanceof UserFacingException) throw (UserFacingException) root;
```

Comment:
Do we really want to compare just the root cause against `UserFacingException`? For example, a few lines above we create a `UserFacingException` with the cause `e`. I'd have thought we'd just want to look if `Exceptions.firstInterestingException(e) instanceof UserFacingException`, and throw that if it is.